DataFormats/EgammaCandidates/src/GsfElectron.cc

```diff
@@ -136,7 +136,13 @@ GsfElectron::GsfElectron(const GsfElectron& electron,
       pixelMatchVariables_(electron.pixelMatchVariables_) {
   trackClusterMatching_.electronCluster = electronCluster;
   //closestCtfTrack_.ctfTrack = closestCtfTrack ;
-  conversionRejection_.partner = conversionPartner;
+  auto& ref = conversionPartner;
+  edm::EDProductGetter const* getter = ref.productGetter();
+  if (getter) {
+    conversionRejection_.partner = TrackPtr(ref.id(), ref.key(), getter);
+  } else {
+    conversionRejection_.partner = TrackPtr(ref.id(), ref.get(), ref.key());
+  }
   //assert(closestCtfTrack==core->ctfTrack()) ;
```

Comment:
This pattern would better be factored out into `DataFormats/Common/interface/RefToPtr.h`

Reply:
I wrote a `refToBaseToPtr` function which I put in `refToBaseToPtr.h` in my version.